src/content/app/tools/blast/state/blast-jobs/blastJobsSelectors.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RootState } from 'src/store';
+
+export const getNextJobId = (state: RootState) =>
```

Comment:
why is it called `getNextJobId`? should it not be `getJobId`

Reply:
Here we are trying to find the id we can use for the upcoming job.

For example, if we have 3 jobs already stored in the state:
```javascript
{
    jobs: [
        { jobId:0 }, 
        { jobId:1 }, 
        { jobId:2 }
   ]
}
```
`jobs.length` will give us 3, which will be used as the id for the upcoming job.

Reply:
I dont think it is for the upcoming job but the current job you just ran. why are we talking about upcoming job? i might have missed something here.

Reply:
As per the jobs already stored in the store, the next job will be the current job we are trying to run.
I say next job here because the current job is not saved to the store yet.

Reply:
I understand your point but still feels it should be just getjobIds. also looking at what its doing, its just returning the length of the object, shouldn't not be returning the value of the jobId instead? (in our case the first one)

Reply:
I am very surprised to see that you are generating job ids on the client side. I would expect the ids to come from the server.

Again, the new redux slice and selector that are responsible for job results do not _really_ belong in the task that this jira ticket is addressing. If you could cleanly isolate these files into a different branch/PR, it would be better. If not, we can accept the current code and update it in subsequent PRs.